File: src/1572. Matrix Diagonal Sum/main.go
package main

func diagonalSum(mat [][]int) int {
	result := 0

	n := len(mat)

	for i := 0; i < n; i++ {
		result += mat[i][i]
		if i != n-i-1 {
			result += mat[i][n-i-1]
		}
	}

	return result
}

File: src/1572. Matrix Diagonal Sum/main.rs
impl Solution {
    pub fn diagonal_sum(mat: Vec<Vec<i32>>) -> i32 {
            let n = mat.len();
            let mut result = 0;

            for i in 0..n {
                result += mat[i][i];
                if i != n - i - 1 {
                    result += mat[i][n-i-1];
                }
            }

            result
    }
}